src/facade/dragonfly_connection.cc

```diff
@@ -769,6 +785,26 @@ auto Connection::IoLoop(util::FiberSocketBase* peer, SinkReplyBuilder* orig_buil
   do {
     FetchBuilderStats(stats_, orig_builder);
 
+    if (migration_request_ && cc_->subscriptions == 0) {
+      ProactorBase* dest = migration_request_;
+
+      if (dispatch_fb_.IsJoinable()) {
+        SendAsync({MigrationRequestMessage{}});
+        dispatch_fb_.Join();
+      }
+
```

Comment:
check: dispatch queue empty

Reply:
ah, here's the catch: You check for subsciptions == 0 above, but what if the command `SUBSCRIBE foo-channel` is contained within the queue 🤓 you should first drain the queue and check only then. I know it sound costly, but we guard against a really really rare case, so it doesn't matter that much

Reply:
nice catch!

Reply:
BTW only thanks to this comment did I realize that Redis has pub/sub API. So far I thought that it was an internal mechanism.